src/test/java/uk/gov/pay/products/resources/ProductResourceIT.java

```diff
@@ -1035,4 +1045,371 @@ public void shouldReturnUsageStats_whenProductAndPaymensExist() {
                 .body("size", is(1))
                 .body("[0].product.external_id", is(product.getExternalId()));
     }
+
+    @Test
+    public void shouldReturn200_whenApiKeyForAPaymentLinkIsUpdated() {
```

Comment:
Not sure we need a variant of this test for each of the different product types: we could probably do with just one and rely on unit tests to make sure we construct the correct request for each of the three types.